### DISCUSSION TITLE
Use Doctum to generate API docs

### DISCUSSION THREAD

**Discussion by williamdes** [php-webdriver/php-webdriver#888] [Ideas]
Hello,

I maintain [Doctum](https://github.com/code-lts/doctum#readme) and I would propose 2 ideas:
- use my GH action to be sure the phpdocs are valid
- use the action or phar or composer package to generate docs and publish them

**Comment by OndraM** [comment]
Hi,
I used sami before to generate the API doc, but it was manuall and and I've done it long time ago. It is pushed in gh-pages branch and published on https://php-webdriver.github.io/php-webdriver/latest/ .

If you are able to make it working again, go for it. I found the old Sami config I used it previously, it was working quite ok.

```php
<?php
use Sami\RemoteRepository\GitHubRemoteRepository;
use Sami\Sami;
use Sami\Version\GitVersionCollection;
use Symfony\Component\Finder\Finder;

$iterator = Finder::create()
    ->files()
    ->name('*.php')
    ->in(__DIR__ . '/lib');

$versions = GitVersionCollection::create(__DIR__ . '/lib')
    ->addFromTags('1.*') // only latest minor version
    ->addFromTags('0.6.0')
    ->add('main', 'main branch')
;

return new Sami($iterator, [
    'title' => 'php-webdriver API',
    'theme' => 'default',
    'build_dir' => __DIR__ . '/sami/build/%version%/',
    'cache_dir' => __DIR__ . '/sami/cache/%version%/',
    'include_parent_data' => true,
    'default_opened_level' => 2,
    'remote_repository'    => new GitHubRemoteRepository('php-webdriver/webdriver', __DIR__),
    'versions' => $versions,
]);
```

What would it require? Can you eg. make it working in your fork and then we will transfer it here?

**Reply by williamdes** [reply]
> What would it require? Can you eg. make it working in your fork and then we will transfer it here?

Barely nothing, I will open a PR tomorrow :)
Just copy pasting https://github.com/code-lts/WarnockPDF/blob/main/.github/workflows/publish.yml and some adjustments will do the trick

> I used sami before to generate the API doc, but it was manuall and and I've done it long time ago. It is pushed in gh-pages branch and published on https://php-webdriver.github.io/php-webdriver/latest/ .

Awesome, le's refresh the generated docs ;p


Here is the script I mainly use to build docs and publish them: https://github.com/code-lts/WarnockPDF/blob/main/scripts/update-docs.sh

Here is an alternative script: https://github.com/williamdes/mariadb-mysql-kbs/blob/main/scripts/sudo-bot/create-pr.sh
